enterprise_access/apps/content_assignments/tasks.py

```diff
@@ -131,3 +131,78 @@ def send_cancel_email_for_pending_assignment(cancelled_assignment_uuid):
+        return
+
+    try:
+        policy = subsidy_policy_model.objects.get(
+            assignment_configuration=assignment.assignment_configuration
+        )
+    except subsidy_policy_model.DoesNotExist:
+        logger.warning(f'policy with assignment config: {assignment.assignment_configuration} does not exist.')
+        return
+
+    learner_content_assignment_action = LearnerContentAssignmentAction(
+        assignment=assignment, action_type=AssignmentActions.REMINDED,
+    )
+    braze_trigger_properties = {}
+    braze_client_instance = BrazeApiClient()
+    lms_client = LmsApiClient()
+    enterprise_customer_uuid = assignment.assignment_configuration.enterprise_customer_uuid
+    enterprise_customer_data = lms_client.get_enterprise_customer_data(enterprise_customer_uuid)
+    admin_emails = [user['email'] for user in enterprise_customer_data['admin_users']]
+    course_metadata = get_content_metadata_for_assignments(
+        policy.catalog_uuid, assignment.assignment_configuration
+    )
+    learner_portal_url = '{}/{}'.format(
+        settings.ENTERPRISE_LEARNER_PORTAL_URL,
+        enterprise_customer_data['slug'],
+    )
+    lms_user_id = assignment.lms_user_id
+    braze_trigger_properties['contact_admin_link'] = braze_client_instance.generate_mailto_link(admin_emails)
+
+    try:
+        recipient = braze_client_instance.create_recipient(
+            user_email=assignment.learner_email,
+            lms_user_id=assignment.lms_user_id,
+        )
+        braze_trigger_properties["organization"] = enterprise_customer_data['name']
+        braze_trigger_properties["course_title"] = assignment.content_title
+        braze_trigger_properties["enrollment_deadline"] = course_metadata['normalized_metadata']['enroll_by_date']
+        braze_trigger_properties["start_date"] = course_metadata['normalized_metadata']['start_date']
+        braze_trigger_properties["course_partner"] = course_metadata['owners'][0]['name']
+        braze_trigger_properties["course_card_image"] = course_metadata['card_image_url']
+        braze_trigger_properties["learner_portal_link"] = learner_portal_url
+
+        logger.info(f'Sending braze campaign message for reminded assignment {assignment}')
+        braze_client_instance.send_campaign_message(
+            settings.BRAZE_ASSIGNMENT_REMINDER_NOTIFICATION_CAMPAIGN,
+            recipients=[recipient],
+            trigger_properties=braze_trigger_properties,
+        )
+        learner_content_assignment_action.completed_at = datetime.now()
+        learner_content_assignment_action.save()
+        return
+    except Exception as exc:  # pylint: disable=broad-exception-caught
+        logger.error(f"Unable to send email for {lms_user_id} due to exception: {exc}")
+        learner_content_assignment_action.error_reason = AssignmentActionErrors.EMAIL_ERROR
+        learner_content_assignment_action.traceback = exc
+        learner_content_assignment_action.save()
```

Comment:
[question/clarification] If the email fails to send, the associated `learner_content_assignment_action` is updated accordingly, but is the `LearnerContentAssignment` record's `state` updated to `errored`? If the assignment record's `state` isn't updated when an error occurs, the UI won't know to display the "Failed: Bad email" status chip.